### DISCUSSION TITLE
Is there a way to prevent deletion in @BeforeDelete hook?

### DISCUSSION THREAD

**Discussion by yf-hk** [mikro-orm/mikro-orm#2315] [Q&A] [answered]
```ts
class User {
  @Property()
  id: string;

  @BeforeDelete()
  async checkDelete() {
    // How to prevent deletion from here?
  }
}
```
Is there any way to stop deletion, like for example, return `false` to stop deletion?

**Comment by B4nan** [comment, marked as answer]
Nope, but it is possible via onFlush subscriber.

**Reply by yf-hk** [reply]
@B4nan Do you think it's feasable to implement a feature, that when the return value of the hook function is `false` (boolean), then cancel delete / update / create? Like in frontend if you catch some event like onClick, returning false will stop the event.

If you think it's OK, I can work on a PR for it.
